src/main/java/sparksoniq/jsoniq/runtime/iterator/RuntimeIterator.java

```diff
@@ -143,7 +146,7 @@ public void write(Kryo kryo, Output output) {
     public void read(Kryo kryo, Input input) {
         this._hasNext = input.readBoolean();
```

Comment:
_hasNext can be set to false as well as _isOpen. _currentDynamicContext can be set to null.

Reply:
Done